### PR TITLE
Item replacement framework

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- A function that lets you register item replacements by class, with or without inheriting classes, and item type

Focus was on performance, should be pretty fast. Intended to be used for medical and the compat free logistics (#7040).

### CODE REVIEW

File: addons/common/functions/fnc_registerItemReplacement.sqf [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+private _firstRun = isNil QGVAR(itemReplacements);
+if (_firstRun) then {
+    GVAR(itemReplacements) = [] call CBA_fnc_createNamespace;
+    GVAR(inheritedReplacements) = [];
+    GVAR(oldItems) = [];
+    ["loadout", _fnc_replaceItems] call CBA_fnc_addPlayerEventHandler;
+};
+
+// Save item replacement
+// $ prefix is used for types (numbers) and replacements with inherition
```
Comment (dedmen) [inline addons/common/functions/fnc_registerItemReplacement.sqf]:
```suggestion
// $ prefix is used for types (numbers) and replacements with inheritance
```

File: addons/common/functions/fnc_registerItemReplacement.sqf [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    private _newItems = _items - GVAR(oldItems);
+    if (_newItems isEqualTo []) exitWith {
+        GVAR(oldItems) = _items;
+    };
+
+    _newItems sort true; // Sort so all items of current class can be replaced at once
+    private _replacements = [];
+    private _cfgWeapons = configFile >> "CfgWeapons"; // Microoptimization
+
+    for "_i" from 0 to count _newItems - 1 do {
```
Comment (dedmen) [inline addons/common/functions/fnc_registerItemReplacement.sqf]: reason that forEach can't be used here?

File: addons/common/functions/fnc_registerItemReplacement.sqf [hunk 3]
```diff
@@ -0,0 +1,115 @@
+#include "script_component.hpp"
+/*
+ * Author: BaerMitUmlaut
+ * Registers an event handler that replaces an item or item type with one or
+ * more other items.
+ *
+ * Arguments:
+ * 0: Item or item type ID to replace <STRING or NUMBER>
```
Comment (dedmen) [inline addons/common/functions/fnc_registerItemReplacement.sqf]: what is "item type ID"?

File: addons/common/functions/fnc_registerItemReplacement.sqf (hunk 2 above)
Comment (BaerMitUmlaut) [reply addons/common/functions/fnc_registerItemReplacement.sqf]: Skipping elements, see L62.

File: addons/common/functions/fnc_registerItemReplacement.sqf (hunk 3 above)
Comment (BaerMitUmlaut) [reply addons/common/functions/fnc_registerItemReplacement.sqf]: https://github.com/acemod/ACE3/blob/master/addons/main/script_macros.hpp#L63-L97

Comment (dedmen) [reply addons/common/functions/fnc_registerItemReplacement.sqf]: Missed Line 47 I guess ^^

**Comment by PabstMirror** [comment]
Merged as part of #7224 (guess git didn't auto-close because of squash?)

**Pull request closed without merging**